R/getTimeDivergence.R

```diff
@@ -0,0 +1,98 @@
+#' Beta diversity calculation between samples at the interval of n time steps
+#'
+#' The dissimilarity (beta distance) is calculated in steps of n between the
+#' samples of subject. "n" can be 1 or greater than 1. The time difference
+#' between the n points is also calculated.
+#'
+#' @param se A
+#' \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#' object or
+#' \code{\link[TreeSummarizedExperiment:TreeSummarizedExperiment-class]{TreeSummarizedExperiment}}
+#' @param sample_field column vector containing the all individuals in
```

Comment:
`sample_field` refers to all individuals, but I am now deleting the `sample_id` field, so it would calculate for all individuals. Maybe the `sample_id` can be an optional parameter in case the dataset is long.

Reply:
I would rather give the user subset the data before using this function if they need to do this for only some limited individuals. The typical use case would be to this for all individuals. 

However, this probably should be implemented so that the function 1) splits the data by individual (or another grouping); 2) performs the operation per individual (or another group); 3) combines the individual results again into a single object.
This would be a robust way of implementing this. Also compare to [lines 141-155 in vignette](https://github.com/microbiome/miaTime/blob/master/vignettes/miaTime.Rmd). The main difference is that the vignette example operates only on `colData` whereas in the current case we need both `colData` and `assay` data -> Can we split the whole SE object by individual (or another group! so that this is not specific to "individuals" but more general to any grouping factor); then calculate the operation per each SE object; then collect the splitted SE objects back to a single SE object?